Create about page and Add data using github api.

lanieerts opened the pull request.

lanieerts committed:
Seperate reducer about comment and modify query.

lanieerts committed:
Modify detail of query for readMoreComments

lanieerts committed:
Modify url of action for createComment.

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Merge remote-tracking branch 'upstream/master'

lanieerts committed:
Add github data in About page using github api.

pjhjohn commented on src/reducers/index.js:
```diff
@@ -2,11 +2,15 @@ import { combineReducers } from 'redux';
   form: formReducer,
-  comment: CommentReducer
+  comment: CommentReducer,
+  contributors: AboutReducer,
```
이거 왜 3줄이나 있졍

lanieerts replied:
그걸 한참 전에 수정해서 올렸어야 했는데 rebase 꼬이면서 push가 안되는 깽판이 났다고 합니다 엉엉.

lanieerts committed:
Combine state of reducers.

lanieerts committed:
Add issue using github api.

pjhjohn commented on src/containers/About/About.js:
```diff
@@ -0,0 +1,54 @@
+class About extends React.Component {
+
+  componentWillMount() {
+    this.props.loadLanguages("/repos/pjhjohn/jsonplaceholder-client/languages");
```
`/repos/pjhjohn/jsonplaceholder-client/` baseUrl로 둬도 되지 않을까염.
그리고 사실상 액션을 나눌거면 저걸 전부 action으로 보내도 될 것 같음.

lanieerts replied:
api가 `GET /repos/:owner/:repo/contributors` <- 이런식이라.. `/repos/pjhjohn/jsonplaceholder-client`를 baseUrl로 넘기고 나머지 부분을 get 안에 넣었더니 정보를 못 들고 오네요.

lanieerts replied:
는 재확인. 그냥 됩니다. `/repos/pjhjohn/jsonplaceholder-client` 부분을 baseUrl로 빼겠습니다.

lanieerts committed:
Modify url for loading github data.
Add simple design using react-bootstrap.

Modify url for loading github data.

Modify parameters of action for load github data.

pjhjohn commented on src/actions/api.js:
```diff
@@ -5,3 +5,7 @@ const BASE_URL = 'http://localhost:3001';
   baseURL: `${BASE_URL}`
 })
+
+export const github = create({
```
일단 요건 api니까 `githubApi` 가 좋을듯

pjhjohn commented on src/_bootstrap.scss:
```diff
@@ -34,7 +34,7 @@
 //@import '~bootstrap-sass/assets/stylesheets/bootstrap/media';
 //@import '~bootstrap-sass/assets/stylesheets/bootstrap/list-group';
-//@import '~bootstrap-sass/assets/stylesheets/bootstrap/panels';
+@import '~bootstrap-sass/assets/stylesheets/bootstrap/panels';
```
이거 이미 [이 커밋](https://github.com/pjhjohn/jsonplaceholder-client/commit/f3043ad8ebc5a611727b7a4db96115cf63a23f33) 에 적용되어있는데, 이거때문에 `Auto Merging` 이 실패하는 것 같네

pjhjohn commented on src/actions/github.js:
```diff
@@ -0,0 +1,38 @@
+import { githubApi } from './api';
+
+export const LOAD_CONTRIBUTORS = 'LOAD_CONTRIBUTORS';
```
굳이 우리가 기존에 쓰던 `READ` 대신 `LOAD` 를 사용한 이유가 있나요?

pjhjohn commented on src/actions/github.js:
```diff
@@ -0,0 +1,38 @@
+export const LOAD_PULL_REQUESTS = 'LOAD_PULL_REQUESTS';
+export const LOAD_ISSUES = 'LOAD_ISSUES';
+
+export function loadContributors(path) {
```
이미 `Contributor`들을 가져오려고 만든 함수인데, 저 `path` 에 다른 값이 들어가게 될 일은 없지 않나요??
그렇다면 아래와 같이 바꾸면 될 것 같아요.

``` js
export function loadContributors() {
  const request = githubApi.get(`/contributors`);
  return {
    type: LOAD_CONTRIBUTORS,
    payload: request
  }
}
```

pjhjohn commented on src/actions/index.js:
```diff
@@ -1,2 +1,3 @@
 export * from './post';
 export * from './comment';
+export * from './github';
```
생각을 다시 해 봤는데, 모든 `githubApi`를 사용하는 `action`&`reducer` 에서 `github` 이라는 이름을 가질 이유가 없는 것 같아요. 같은 논리라면 우리는 `api`를 쓰니까 모두 한 파일에 넣어야 한다는 뜻이 되거든요.

`api`를 모두 사용하지만 **용도에 따라서 `action`&`reducer` 이름을 지었기 때문에** about 페이지에서 사용하는 부분만 `about.js`로 `action`&`reducer` 이름을 정하면 될 것이고, 만약 `githubApi`를 다른 페이지에서 사용해야 할 이유가 있다면 따로 만들어주면 되겠지요 :D

pjhjohn commented on src/containers/About/About.js:
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Inspector } from 'react-inspector';
```
Component로 분리하는게 어떨까요?
지금은 Inspector를 사용하지만, 추후에는 해당 뷰를 Inspector에서 더 발전시킬 수도 있을 테니까요.

React를 사용하는 가장 큰 장점 중 하나가 **"Component 이름을 이용한 의미(Semantically) 단위의 계층 구조 표현"** 아닐까요?
마냥 Inspector를 넣고 코드를 보는 사람이 "여기에 어디보자...그러니까 여기 props로 음... `issues` 가 들어가니까 이건 이슈를 가져와서 뿌려주는 뷰가 될 거야 - 라고 이해하는건 또 하나의 노동이 될 것 같아요. (현 프로젝트의 `PostDetail` 과 `Comment` 컴포넌트 모두 Inspector를 사용하는건데 굳이 Component로 분리시켜 놓았죠)

pjhjohn replied:
Component로 분리하는게 어떨까요?
지금은 Inspector를 사용하지만, 추후에는 해당 뷰를 Inspector에서 더 발전시킬 수도 있을 테니까요.

React를 사용하는 가장 큰 장점 중 하나가 **"Component 이름을 이용한 의미(Semantically) 단위의 계층 구조 표현"** 아닐까요?
마냥 Inspector를 넣고 코드를 보는 사람이 "여기에 어디보자...그러니까 여기 props로 음... `issues` 가 들어가니까 이건 이슈를 가져와서 뿌려주는 뷰가 될 거야 - 라고 이해하는건 또 하나의 노동이 될 것 같아요. (현 프로젝트의 `PostDetail` 과 `Comment` 컴포넌트 모두 Inspector를 사용하는건데 굳이 Component로 분리시켜 놓았죠)

또한, `containers/` 자체가 `pure-react` 보다는 `redux`가 붙은, 뭔가의 동작을 추가하는 통로로 사용하기로 했었는데요 (굳이 하나로 해도 될 것도 `connect` 부분 등으로 따로 빼서 만들었죠), 이렇게 하면 **컴포넌트 라이브러리의 사용은 되도록이면 `components/` 에서 사용하게 되는** 부수적인 효과가 있습니다. (이 부분은 React-Bootstrap으로 만들어진 현재 컨테이너에 있는 뷰 또한 일부 추상화되어 컴포넌트쪽으로 가게 될 것임을 암시합니다)

pjhjohn commented on src/containers/About/About.scss:
```diff
@@ -0,0 +1 @@
+
```
더 좋은 인스트럭션을 제공하기 위해 TODO 주석을 넣어주세요

pjhjohn commented on src/containers/Header/Header.js:
```diff
@@ -27,7 +27,7 @@ class Header extends React.Component {
           </Nav>
           <Nav pullRight>
-            <LinkContainer to={{ pathname: '/albums' }}>
+            <LinkContainer to={{ pathname: '/about' }}>
```
Thanks.

pjhjohn commented on src/reducers/github.js:
```diff
@@ -0,0 +1,18 @@
+import { LOAD_CONTRIBUTORS, LOAD_LANGUAGES, LOAD_PULL_REQUESTS, LOAD_ISSUES } from './../actions';
+
+const INITIAL_STATE = { list: [], detail: {} };
```
막 복붙하지 맙시다.

pjhjohn commented on src/reducers/index.js:
```diff
@@ -2,11 +2,13 @@ import { combineReducers } from 'redux';
 import { reducer as formReducer } from 'redux-form';
 import PostReducer from './post';
 import CommentReducer from './comment';
+import AboutReducer from './github';
```
이것 또한 위에 제시한 `github` -> `about` 의 근거 중 하나입니다.
일관성이 깨지는 부분이 이렇게 생기거든요.

pjhjohn commented:
중복된 문제의 경우 굳이 재차 언급하지 않은 경우도 있습니다.
수정 후 자신이 생길 때 까지 셀프 리뷰를 해 보시고 PR해주세요.
